src/CMakeLists.txt

```diff
@@ -143,6 +143,17 @@ if(APPLE AND (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" OR CMAKE_CXX_COMPILER_
         list(REMOVE_DUPLICATES CMAKE_CXX_IMPLICIT_LINK_LIBRARIES)
 endif()
 
+# Check for base threading library
+set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
+set(THREADS_PREFER_PTHREAD_FLAG TRUE)
+find_package(Threads REQUIRED)
+if(${CMAKE_THREAD_LIBS_INIT} MATCHES "-pthread")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
+elseif(CMAKE_THREAD_LIBS_INIT)
```

Comment:
Also, we specifically want pthreads, and this will not fail if it does not find pthreads, but finds something else.
Something like this is needed https://github.com/darktable-org/rawspeed/blob/5db36ca57ede56829cb90ec68a528d593ebf4cdc/RawSpeed/CMakeLists.txt#L4-L6